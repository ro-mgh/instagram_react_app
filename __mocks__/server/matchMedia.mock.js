Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
// window.matchMedia = jest.fn().mockImplementation((query) => {
//   return {
//     matches: true,
//     media: query,
//     onchange: null,
//     addListener: jest.fn(), // deprecated
//     removeListener: jest.fn(), // deprecated
//     addEventListener: jest.fn(),
//     removeEventListener: jest.fn(),
//     dispatchEvent: jest.fn(),
//   };
// });
