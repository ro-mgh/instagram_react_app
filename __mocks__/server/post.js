const post = {
  author: {
    avatar:
      "https://insta-project.s3-ap-northeast-2.amazonaws.com/waladPiWs1XukX995ZBlSNODoTH3.jpeg",
    id: "waladPiWs1XukX995ZBlSNODoTH3",
    username: "burningman",
  },
  comments: [],
  createdAt: "2021-02-19T09:04:02.437Z",
  id: 49,
  image:
    "https://insta-project.s3-ap-northeast-2.amazonaws.com/waladPiWs1XukX995ZBlSNODoTH3&&1613725435570.jpeg",
  likes: [],
  text: null,
};

export const _post = post;
