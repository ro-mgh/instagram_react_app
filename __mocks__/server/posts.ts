const posts = [
  {
    createdAt: "2021-02-07T10:35:12.443Z",
    id: 27,
    image:
      "https://insta-project.s3-ap-northeast-2.amazonaws.com/hPpgfQK9i3MC1ATiW1BHP7jIDai1&&1612694084227.jpeg",
  },
  {
    createdAt: "2021-02-04T16:33:31.034Z",
    id: 26,
    image:
      "https://insta-project.s3-ap-northeast-2.amazonaws.com/hPpgfQK9i3MC1ATiW1BHP7jIDai1&&1612456403166.jpeg",
  },
  {
    createdAt: "2021-02-04T16:33:18.032Z",
    id: 25,
    image:
      "https://insta-project.s3-ap-northeast-2.amazonaws.com/hPpgfQK9i3MC1ATiW1BHP7jIDai1&&1612456392129.jpeg",
  },
];

export const _posts = posts;
