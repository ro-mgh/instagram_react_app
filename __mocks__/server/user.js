export const _user = {
  id: "123",
  avatar: "",
  username: "test",
  name: "testName",
};
