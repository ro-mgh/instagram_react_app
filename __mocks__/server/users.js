export const _users = {
  SOMEID: {
    id: "id",
    username: "test",
    name: "testName",
    avatar: "",
    followingIds: [],
    Following: [],
    posts: [],
  },
};
