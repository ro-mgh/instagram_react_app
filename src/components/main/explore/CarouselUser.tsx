import React, { useState } from "react";
import Avatar from "@material-ui/core/Avatar";
import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
import firebase from "../../../services/firebase";
import { useDispatch } from "react-redux";
import { exploreUsers } from "../../../store/actions/exploreUsers";
import { Link } from "react-router-dom";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    medium: {
      width: theme.spacing(7),
      height: theme.spacing(7),
    },
  })
);

const CarouselUser = (props) => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const [isFollow, setFollow] = useState(false);

  const handleFollow = async () => {
    setFollow(true);
    firebase
      .auth()
      .currentUser.getIdToken(/* forceRefresh */ true)
      .then(async function (idToken) {
        try {
          const response = await fetch(
            "http://localhost:3000/follow/" + props.id,
            {
              method: "post",
              headers: {
                "Content-type": "application/json",
                Authorization: "Bearer " + idToken,
              },
            }
          );
          if (response.ok) {
            // const jsonResponse = await response.json();
          } else {
            console.error("error");
          }
        } catch (e) {
          console.error(e);
        }
      })
      .catch(function (error) {
        // Handle error
        console.log(error);
      });
  };

  const handleUnfollow = async () => {
    setFollow(false);
    firebase
      .auth()
      .currentUser.getIdToken(/* forceRefresh */ true)
      .then(async function (idToken) {
        try {
          const response = await fetch(
            "http://localhost:3000/follow/" + props.id,
            {
              method: "delete",
              headers: {
                "Content-type": "application/json",
                Authorization: "Bearer " + idToken,
              },
            }
          );
          if (response.ok) {
            // const jsonResponse = await response.json();
          } else {
            console.error("error");
          }
        } catch (e) {
          console.error(e);
        }
      })
      .catch(function (error) {
        // Handle error
        console.log(error);
      });
  };

  return (
    <div key={props.id} className="carousel-wrapper">
      <div className="carousel-user-wrapper">
        <Link to={"/profile/" + props.id}>
          <div className="carousel-user-avatar">
            <Avatar alt="A" src={props.avatar} className={classes.medium} />
          </div>
        </Link>
        <div className="post-header-username">
          <Link to={"/profile/" + props.id}>
            <div className="username-font">{props.username}</div>
          </Link>
        </div>
        {isFollow ? (
          <button className="follow-button" onClick={handleUnfollow}>
            Unfollow
          </button>
        ) : (
          <button className="follow-button" onClick={handleFollow}>
            Follow
          </button>
        )}
      </div>
    </div>
  );
};

export default CarouselUser;
