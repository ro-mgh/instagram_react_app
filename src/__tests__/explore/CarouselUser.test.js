import { unmountComponentAtNode } from "react-dom";
import React from "react";
import { render, screen } from "@testing-library/react";
import { _user } from "../../../__mocks__/server/user";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

import CarouselUser from "../../components/main/explore/CarouselUser";

import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureStore([]);

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("rendering CarouselUser component", () => {
  it("renders CarouselUser component", () => {
    const history = createMemoryHistory();

    let store = mockStore({
      store: "",
    });

    const component = render(
      <Provider store={store}>
        <Router history={history}>
          <CarouselUser {..._user} />
        </Router>
      </Provider>
    );

    // screen.debug();
    expect(component.container).toMatchSnapshot();
    expect(screen.getByTestId("carousel-user")).toBeInTheDocument();
  });
});
