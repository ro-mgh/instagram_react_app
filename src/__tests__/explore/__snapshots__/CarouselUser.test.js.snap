// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering CarouselUser component renders CarouselUser component 1`] = `
<div>
  <div
    class="carousel-wrapper"
    data-testid="carousel-user"
  >
    <div
      class="carousel-user-wrapper"
    >
      <a
        href="/profile/123"
      >
        <div
          class="carousel-user-avatar"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-medium-1 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
      </a>
      <div
        class="div-username-explore"
      >
        <a
          href="/profile/123"
        >
          <div
            class="username-font  username-font-explore"
          >
            test
          </div>
        </a>
      </div>
      <button
        class="follow-button"
      >
        Follow
      </button>
    </div>
  </div>
</div>
`;
